### ISSUE TITLE
"flutter run -d chrome" doesn't work with the downloaded zip versions of Flutter unless you run `flutter update-packages` first

### ISSUE THREAD

**Issue by DanTup** [flutter/flutter#41681]
Moving from https://github.com/Dart-Code/Dart-Code/issues/2012

If you grab one of the zip releases from here:

https://flutter.dev/docs/development/tools/sdk/releases?tab=macos

Then run `flutter run -d chrome`, it'll fail with:

```
Error: Error when reading '/Users/dantup/Desktop/flutter-dev/packages/flutter_tools/.packages': No such file or directory
```

Running `flutter update-packages` with that version of Flutter causes the packages file to be created, and the error goes away. It seems like this should happen automatically (or the zip file should include the `.packages` file).

**Comment by escamoteur** [comment]
@DanTup shouldn't you run flutter doctor in any case?

**Comment by maheshmnj** [comment]
Hi @DanTup, I verified by downloading the latest stable release 3.0.5 and I am unable to reproduce the issue. Could you please confirm if you are still experiencing the above error?

### logs
```
mahesh@Maheshs-MacBook-Air-M1 Desktop % cd sample 
mahesh@Maheshs-MacBook-Air-M1 sample % ~/Downloads/flutter/bin/flutter run -d chrome
Running "flutter pub get" in flutter_tools...                      15.1s
Launching lib/main.dart on Chrome in debug mode...
Waiting for connection from debug service on Chrome...              9.3s
This app is linked to the debug service: ws://127.0.0.1:59420/Fsy3Ek_RLis=/ws
Debug service listening on ws://127.0.0.1:59420/Fsy3Ek_RLis=/ws

💪 Running with sound null safety 💪

🔥  To hot restart changes while running, press "r" or "R".
For a more detailed help message, press "h". To quit, press "q".

An Observatory debugger and profiler on Chrome is available at: http://127.0.0.1:59420/Fsy3Ek_RLis=
Flutter Web Bootstrap: Programmatic
hello
The Flutter DevTools debugger and profiler on Chrome is available at: http://127.0.0.1:9101?uri=http://127.0.0.1:59420/Fsy3Ek_RLis=
the Dart compiler exited unexpectedly.
mahesh@Maheshs-MacBook-Air-M1 sample % ~/Downloads/flutter/bin/flutter --version    
Flutter 3.0.5 • channel stable • https://github.com/flutter/flutter.git
Framework • revision f1875d570e (7 days ago) • 2022-07-13 11:24:16 -0700
Engine • revision e85ea0e79c
Tools • Dart 2.17.6 • DevTools 2.12.2
mahesh@Maheshs-MacBook-Air-M1 sample % 
```

**Comment by DanTup** [comment]
Yep, this seems solved. `.packages` has gone away in favour of .dart_tool/package_config.json` but it seems to be created when I first run the command above.
